metrics.nim

```diff
@@ -938,14 +938,24 @@ when defined(metrics):
       # TODO: parse the output of `GC_getStatistics()` for more stats
 
       when defined(nimTypeNames) and declared(dumpHeapInstances):
+        var heapNames = initTable[cstring, int]()
         var heapSizes: array[10, (cstring, int)]
+
+        # de-duplicate return values, labels should be unique
         for data in dumpHeapInstances():
+          if heapNames.contains(data.name):
+            heapNames[data.name] += data.sizes
+          else:
+            heapNames[data.name] = data.sizes
```

Comment:
This is heap allocation which should be avoided while looping over that iterator.

Reply:
Oh, because tables grow as you put stuff into it?

Reply:
Then I guess the only solution is to have a pre-defined size of array and just add yet another loop within the other loops to check for the name.

Reply:
Actually, what happens if you allocate during the loop? Does it break? Or does it just mess with the data?